### PR TITLE
Add Sundials 5.2.0 & 5.3.0

### PR DESCRIPTION
**Pull request by vvilpas**
Specify library name and version:  **sundials/5.2.0 & 5.3.0** 

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sundials/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from conans import CMake, ConanFile, tools
+import os
+
+
+class SundialsConan(ConanFile):
+    name = "sundials"
+    license = "BSD 3-Clause"
```
Comment (uilianries) [inline recipes/sundials/all/conanfile.py:7]:
```suggestion
    license = "BSD-3-Clause"
```

File: recipes/sundials/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "build_arkode": [True, False],
+               "build_cvode": [True, False],
+               "build_cvodes": [True, False],
+               "build_ida": [True, False],
+               "build_idas": [True, False],
+               "build_kinsol": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": False,
```
Comment (uilianries) [inline recipes/sundials/all/conanfile.py:25]:
```suggestion
                       "fPIC": True,
```

File: recipes/sundials/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                       "build_arkode": True,
+                       "build_cvode": True,
+                       "build_cvodes": True,
+                       "build_ida": True,
+                       "build_idas": True,
+                       "build_kinsol": True}
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake"
+    short_paths = True
```
Comment (uilianries) [inline recipes/sundials/all/conanfile.py:34]:
```suggestion
```

File: recipes/sundials/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```
Comment (uilianries) [inline recipes/sundials/all/conanfile.py:59]:
```suggestion
```

Automatically configured by CMake helper

File: recipes/sundials/all/conanfile.py (hunk 3 above)
Comment (vvilpas) [reply recipes/sundials/all/conanfile.py:34]: It doesn't compile in windows without this, due to very long names.

**Comment by vvilpas** [comment]
I have removed the tests and put a minimum example to test some includes and linking.
Addressed all other comments except short_paths, beacuse it doesn't compile in windows due to long names issue.

File: recipes/sundials/all/conanfile.py (hunk 3 above)
Comment (uilianries) [reply recipes/sundials/all/conanfile.py:34]: fair enough

**Pull request merged**